scripts/use_node

```diff
@@ -0,0 +1,97 @@
+#!/bin/sh
+
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+#
+# Any modifications Copyright OpenSearch Contributors. See
+# GitHub history for details.
+#
+
+# This script will find the appropriate Node.js runtime binary and execute it with any
+# parameters passed in.
+#
+# Set a variable named OSD_USE_NODE_JS_FILE_PATH to have it prefixed with OSD_HOME and executed
+# Example: SET OSD_USE_NODE_JS_FILE_PATH=\src\cli\dist
+#
+# NODE_OPTIONS is built using config/node.options and overridden by any previously set NODE_OPTIONS.
+# To pass in any specific defaults that can be overridden by both of them, use OSD_NODE_OPTS_PREFIX.
+
+SCRIPT="$0"
+
+UNAME=$(uname -s)
+if [ $UNAME = "FreeBSD" ]; then
+  OS="freebsd"
+elif [ $UNAME = "Darwin" ]; then
+  OS="darwin"
+else
+  OS="other"
+fi
+
+# SCRIPT may be an arbitrarily deep series of symlinks. Loop until we have the concrete path.
+while [ -h "$SCRIPT" ] ; do
+  ls=$(ls -ld "$SCRIPT")
+  # Drop everything prior to ->
+  link=$(expr "$ls" : '.*-> \(.*\)$')
+  if expr "$link" : '/.*' > /dev/null; then
+    SCRIPT="$link"
+  else
+    SCRIPT=$(dirname "$SCRIPT")/"$link"
+  fi
+done
+
+# Get an absolute path for OSD_HOME
+OSD_HOME="$(cd "$(dirname "${SCRIPT}")/.."; pwd)"
+CONFIG_DIR=${OSD_PATH_CONF:-"OSD_HOME/config"}
+
+# Places to look for the Node.js binary in order: OSD_NODE_HOME > NODE_HOME > bundled with OSD > system-wide
+if [ ! -z "$OSD_NODE_HOME" ]; then
+  NODE="$OSD_NODE_HOME/bin/node"
+  NODE_ERROR_MSG="in OSD_NODE_HOME"
+  NODE_ERROR_SHOW=true
+elif [ ! -z "$NODE_HOME" ]; then
+  NODE="$NODE_HOME/bin/node"
+  NODE_ERROR_MSG="in NODE_HOME"
+  NODE_ERROR_SHOW=true
+else
+  NODE="$OSD_HOME/node/bin/node"
+  NODE_ERROR_MSG="bundled with OpenSearch Dashboards"
+  # A bin folder at the root is only present in release builds that have a bundled Node.js binary
+  if [ -x "OSD_HOME/bin" ]; then
+    NODE_ERROR_SHOW=true
+  fi
+fi
+
+if [ -x "$NODE" ]; then
+  # Node.js binary was found where it was expected; no need to show an error
+  NODE_ERROR_SHOW=
+elif [ $OS = "freebsd" ]; then
+  NODE="${LOCALBASE}/bin/node"
+else
+  NODE="$(which node)"
```

Comment:
`which` is not always the default installed pkg on docker.
This might impact our docker image and need to confirm if it exist.
Also on our ci images for integTests and such.

Reply:
A verification of which before running the command is good.

Reply:
Done.